homeassistant/components/husqvarna_automower/__init__.py

```diff
@@ -53,6 +60,10 @@ async def async_setup_entry(hass: HomeAssistant, entry: AutomowerConfigEntry) ->
 
     coordinator = AutomowerDataUpdateCoordinator(hass, automower_api, entry)
     await coordinator.async_config_entry_first_refresh()
+    available_devices = list(coordinator.data)
+    if TYPE_CHECKING:
+        assert coordinator.config_entry is not None
```

Comment:
Hold up, instead of doing this, rather just overwrite the type in the coordinator since it can't be `None`

```py
class AutomowerDataUpdateCoordinator(DUC[..]):

    config_entry: ConfigEntry

    def __init__(self):
```